org.eclipse.scanning.connector.epics/src/org/eclipse/scanning/connector/epics/EpicsV4ConnectorService.java

```diff
@@ -159,18 +160,19 @@ public void subscribe(IMalcolmDevice<?> device, MalcolmMessage msg, IMalcolmList
 
 	@Override
 	public void subscribeToConnectionStateChange(IMalcolmDevice<?> device, IMalcolmListener<Boolean> listener)
-			throws MalcolmDeviceException  {
+			throws MalcolmDeviceException {
 
 		try {
-			PvaClientChannel pvaChannel = pvaClient.createChannel(device.getName(),"pva");
-	        pvaChannel.issueConnect();
-	        Status status = pvaChannel.waitConnect(0); // Wait forever for this connection.
-	        if(!status.isOK()) {
-			String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": " + status.getMessage() + ")";
-			logger.error(errMEssage);
-			throw new Exception(errMEssage);
-	        }
-	        pvaChannel.setStateChangeRequester(new StateChangeRequester(listener));
+			PvaClientChannel pvaChannel = pvaClient.createChannel(device.getName(), "pva");
+			pvaChannel.issueConnect();
+			Status status = pvaChannel.waitConnect(0); // Wait forever for this connection.
+			if (!status.isOK()) {
+				String errMEssage = "Failed to connect to device '" + device.getName() + "' (" + status.getType() + ": "
+						+ status.getMessage() + ")";
+				logger.error(errMEssage);
+				throw new Exception(errMEssage);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)